# SPaRTAN
The developmental pathways and functions of specialized cell types are dependent on the complex interplay between signaling and transcriptional networks. We present SPaRTAN (Single-cell Proteomic and RNA based Transcription factor Activity Network), a computational method to link cell-surface receptors to transcription factors (TFs) by exploiting Cellular Indexing of Transcriptomes and Epitopes by Sequencing (CITE-seq) datasets with cis-regulatory information. SPaRTAN is applied to peripheral blood mononuclear cells (PBMCs) to predict the coupling of signaling receptors with cell context-specific TF activities. The predictions are validated by flow cytometry analyses. SPaRTAN is then used to analyze the signaling coupled TF activity states of tumor infiltrating CD8+ T cells in malignant peritoneal and pleural mesotheliomas. SPaRTAN greatly enhances the utility of CITE-seq datasets to probe signaling coupled TF networks that regulate developmental or functional transitions in cellular states.

![Diagram](https://github.com/DRP60/SPaRTAN/blob/main/data/diagram.png)

Briefly, our model views expression of surface proteins as a proxy of their activities; signaling emanating from these proteins converges on particular TFs, whose activities, in turn, regulate the expression of their target genes. Specifically, we use a regularized bilinear regression algorithm called affinity regression (AR), a general statistical framework for any problem where the observed data can be explained as interactions (**W**) between two kinds of inputs, to establish an interaction matrix between surface proteins (**P**) and TFs (**D**)  that predicts target gene expression (**Y**). 

Since the model captures statistical relationships between surface proteins, TFs, and gene expression. We can use the trained interaction matrix to obtain different views of a CITE-seq data set; e.g., to predict TF activity from a cell's surface protein expression profile or to predict surface protein expression from a cellâ€™s gene expression profile.  Intuitively, information flows down from observed surface protein levels through the learned interaction matrix to infer TF activities and observed mRNA expression levels or propagates up through the TF target-gene edges and interaction network to infer surface protein expression. 

There is currently an implementation in Python. For more details and installation instructions on running SPaRTAN, please see the tutorial:
* [Run SPaRTAN in Python](https://github.com/osmanbeyoglulab/SPaTRAN2/tree/main/SPaRTAN_python)
